src/Workspaces/Core/Portable/Workspace/Solution/ProjectId.cs

```diff
@@ -28,8 +28,12 @@ private ProjectId(string debugName)
             _debugName = debugName;
         }
 
-        internal ProjectId(Guid guid, string debugName)
+        public ProjectId(Guid guid, string debugName = null)
```

Comment:
Regardless of what we pick, IMO we should never have *both* publicly accessible constructors and static factory methods - people will always just use the constructor.

I think a named method that says "CreateFromSerialized" (or better) would be good here.